plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestIcebergPolarisCatalogCaseInsensitiveMapping.java

```diff
@@ -0,0 +1,204 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static org.apache.iceberg.rest.auth.OAuth2Properties.CREDENTIAL;
+import static org.apache.iceberg.rest.auth.OAuth2Properties.SCOPE;
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@TestInstance(PER_CLASS)
+final class TestIcebergPolarisCatalogCaseInsensitiveMapping
+        extends AbstractTestQueryFramework
+{
+    private static final SessionContext SESSION_CONTEXT = new SessionContext("dummy", null, null, ImmutableMap.of(), "user");
+    private RESTSessionCatalog icebergCatalog;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        File warehouseLocation = Files.newTemporaryFolder();
+        TestingPolarisCatalog polarisCatalog = closeAfterClass(new TestingPolarisCatalog(warehouseLocation.getPath()));
+
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put(CatalogProperties.URI, polarisCatalog.restUri() + "/api/catalog")
+                .put(CatalogProperties.WAREHOUSE_LOCATION, TestingPolarisCatalog.WAREHOUSE)
+                .put(CREDENTIAL, polarisCatalog.oauth2Credentials())
+                .put(SCOPE, "PRINCIPAL_ROLE:ALL")
+                .buildOrThrow();
+
+        RESTSessionCatalog icebergCatalogInstance = new RESTSessionCatalog(
+                config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build(), null);
+        icebergCatalogInstance.initialize("test_catalog", properties);
+
+        icebergCatalog = icebergCatalogInstance;
+        closeAfterClass(icebergCatalog);
+
+        return IcebergQueryRunner.builder()
+                .setBaseDataDir(Optional.of(warehouseLocation.toPath()))
+                .addIcebergProperty("iceberg.catalog.type", "rest")
+                .addIcebergProperty("iceberg.rest-catalog.uri", polarisCatalog.restUri() + "/api/catalog")
+                .addIcebergProperty("iceberg.rest-catalog.warehouse", TestingPolarisCatalog.WAREHOUSE)
+                .addIcebergProperty("iceberg.rest-catalog.security", "OAUTH2")
+                .addIcebergProperty("iceberg.rest-catalog.oauth2.credential", polarisCatalog.oauth2Credentials())
+                .addIcebergProperty("iceberg.rest-catalog.oauth2.scope", "PRINCIPAL_ROLE:ALL")
+                .addIcebergProperty("iceberg.register-table-procedure.enabled", "true")
+                .addIcebergProperty("iceberg.rest-catalog.case-insensitive-name-matching", "true")
+                .build();
+    }
+
+    @Test
+    void testCaseInSensitiveMatching()
```

Comment:
nit: `InSensitive` → `Insensitive` for consistency with `iceberg.rest-catalog.case-insensitive-name-matching` naming.